test/e2e/flowaggregator_test.go

```diff
@@ -691,9 +695,47 @@ func getUnit64FieldFromRecord(t *testing.T, record string, field string) uint64
 	return 0
 }
 
-// TODO: Add a test that checks the functionality of Pod-To-External flow.
-func checkFlowType(t *testing.T, record string, flowType uint8) {
-	assert.Containsf(t, record, fmt.Sprintf("flowType: %d", flowType), "Record does not have correct flowType")
+func getExternalServerIPv4(t *testing.T, server string) string {
+	ips, _ := net.LookupIP(server)
+	var ipv4 string
+	for _, ip := range ips {
+		if ip.To4() != nil {
+			ipv4 = ip.String()
+		}
+	}
+	return ipv4
+}
+
+func getCollectorOutput(t *testing.T, srcIP string, dstIP string, timeStart time.Time, checkAllRecords bool) string {
+	err := wait.PollImmediate(500*time.Millisecond, aggregatorInactiveFlowRecordTimeout, func() (bool, error) {
+		// `pod-running-timeout` option is added to cover scenarios where ipfix flow-collector has crashed after being deployed
+		rc, collectorOutput, _, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl logs --since=%v --pod-running-timeout=%v ipfix-collector -n antrea-test", time.Since(timeStart).String(), aggregatorInactiveFlowRecordTimeout.String()))
+		if err != nil || rc != 0 {
+			return false, err
+		}
+		// Checking all the data records correspondent to the iperf flow are received
+		if checkAllRecords {
+			recordSlices := getRecordsFromOutput(collectorOutput)
+			for _, record := range recordSlices {
+				exportTime := int64(getUnit64FieldFromRecord(t, record, "flowEndSeconds"))
+				if strings.Contains(record, srcIP) && strings.Contains(record, dstIP) {
+					if exportTime >= timeStart.Unix()+iperfTimeSec {
+						return true, nil
+					}
+				}
+			}
+			return false, nil
+		} else {
+			return strings.Contains(collectorOutput, srcIP) && strings.Contains(collectorOutput, dstIP), nil
+		}
+	})
+	require.NoErrorf(t, err, "IPFIX collector did not receive the expected records and timed out with error: %v", err)
+
+	rc, collectorOutput, _, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl logs --since=%v ipfix-collector -n antrea-test", time.Since(timeStart).String()))
```

Comment:
Is it duplicate with the process in `wait.PollImmediate`? Maybe you can declare `collectorOutput` out of `wait.PollImmediate` so it can be kept on success and be returned directly?